subprojects/dependency-management/src/main/java/org/gradle/api/internal/artifacts/ivyservice/ivyresolve/parser/GradleModuleMetadataParser.java

```diff
@@ -228,6 +228,15 @@ private void consumeVariant(JsonReader reader, MutableModuleComponentResolveMeta
         reader.endObject();
 
         MutableComponentVariant variant = metadata.addVariant(variantName, attributes);
+        variant.setAvailableExternally(availableExternally);
+        if (availableExternally) {
```

Comment:
I wonder if these added tests will break on anything published so far. Note that I am not saying we should not do them!

Reply:
Can you clarify what you mean/think could break?

Reply:
This is the parser and the checks below are new AFAICT. So if anyone published an `available-at` variant with constraints or files, it will no longer be parsed.

Reply:
Right I see what you mean. Indeed the assertions can fail. However if they do it means that the published files didn't follow the spec.